packages/relay/src/lib/services/clientService/index.ts

```diff
@@ -0,0 +1,276 @@
+   * @private
+   */
+  private logger: Logger;
+  private readonly register: Registry;
+  private clientResetCounter: Counter;
+
+  private chainIds = {
+    mainnet: 0x127,
+    testnet: 0x128,
+    previewnet: 0x129,
+  };
+
+  /**
+   * @param {Logger} logger
+   * @param {Registry} register
+   */
+  constructor(logger: Logger, register: Registry) {
+    dotenv.config({ path: findConfig('.env') || '' });
+
+    this.logger = logger;
+
+    const hederaNetwork: string = (process.env.HEDERA_NETWORK || '{}').toLowerCase();
+    this.clientMain = this.initClient(logger, hederaNetwork);
+    this.client = this.initSDKClient(logger, register);
+
+    const currentDateNow = Date.now();
+    this.transactionCount = parseInt(process.env.CLIENT_TRANSACTION_RESET!) || 0;
+    this.resetDuration = currentDateNow + parseInt(process.env.CLIENT_DURATION_RESET!) || 0;
+    this.errorCount = parseInt(process.env.CLIENT_ERROR_RESET!) || 0;
+    this.isEnabled = true;
+    this.isTimeResetDisabled = this.resetDuration === currentDateNow;
+
+    if (this.transactionCount === 0 && this.errorCount === 0 && this.isTimeResetDisabled) {
+      this.isEnabled = false;
+    }
+    this.shouldReset = false;
+
+    this.register = register;
+
+    const metricCounterName = 'rpc_relay_client_service';
+    register.removeSingleMetric(metricCounterName);
+    this.clientResetCounter = new Counter({
+      name: metricCounterName,
+      help: 'Relay Client Service',
+      registers: [register],
+      labelNames: ['transactions', 'duration', 'errors'],
+    });
+  }
+
+  /**
+   *  Decrement transaction counter. If 0 is reached, reset the client. Check also if resetDuration has been reached and reset the client, if yes.
+   */
+  private decrementTransactionCounter() {
+    if (this.transactionCount == 0) {
+      return;
+    }
+
+    this.transactionCount--;
+    if (this.transactionCount <= 0) {
+      this.shouldReset = true;
+    }
+  }
+
+  /**
+   *  Decrement error encountered counter. If 0 is reached, reset the client. Check also if resetDuration has been reached and reset the client, if yes.
+   */
+  public decrementErrorCounter() {
+    if (!this.isEnabled || this.errorCount == 0) {
+      return;
+    }
+
+    this.errorCount--;
+    if (this.errorCount <= 0) {
+      this.shouldReset = true;
+    }
+  }
+
+  private checkResetDuration() {
+    if (this.isTimeResetDisabled) {
+      return;
+    }
+    
+    if (this.resetDuration < Date.now()) {
+      this.shouldReset = true;
+    }
+  }
+
+  /**
+   * Reset the main client, SDK Client and reset all counters.
+   */
+  private resetClient() {
+    this.clientResetCounter
+      .labels(this.transactionCount.toString(), this.resetDuration.toString(), this.errorCount.toString())
+      .inc(1);
+
+    const hederaNetwork: string = (process.env.HEDERA_NETWORK || '{}').toLowerCase();
+    this.clientMain = this.initClient(this.logger, hederaNetwork);
```

Comment:
I think is the same case `this.clientMain` I think we should not reInit it every time we reset the sdkClient, the `Client` only has configuration about the network, and it won't change. we should not `re-init` the client.